cloudslang-runtime/src/test/java/io/cloudslang/lang/runtime/steps/ActionStepsTest.java

```diff
@@ -135,6 +136,75 @@ public void doJavaActionWrongClassTest() {
         Assert.assertNotNull(actionEndEvent);
     }
 
+    @Test
+    public void doActionPythonActionCheckCallArgumentsOnEvent() {
+        //prepare doAction arguments
+        RunEnvironment runEnv = new RunEnvironment();
+        ExecutionRuntimeServices runtimeServices = new ExecutionRuntimeServices();
+
+        Map<String, Serializable> callArguments = new HashMap<>();
+        callArguments.put("index", 1);
+        runEnv.putCallArguments(callArguments);
+
+        String userPythonScript = "print \"hello\"";
+
+        //invoke doAction
+        actionSteps.doAction(runEnv,new HashMap<String, Object>(), PYTHON, "", "", runtimeServices, userPythonScript, 2L, "operationName");
+
+        Collection<ScoreEvent> events = runtimeServices.getEvents();
+
+        Assert.assertFalse(events.isEmpty());
+        ScoreEvent eventActionStart = null;
+        for(ScoreEvent event:events){
+            if(event.getEventType().equals(ScoreLangConstants.EVENT_ACTION_START)){
+                eventActionStart = event;
+            }
+        }
+        assert eventActionStart != null;
+        Map<String, Serializable> data = (HashMap)eventActionStart.getData();
+        for (Map.Entry<String, Serializable> entry : data.entrySet()) {
```

Comment:
how I would refactor this section (e.g. no need for `for`):

- cast `.getData()` to `LanguageEventData`
- update `LanguageEventData` with getter / setter for `CALL_ARGUMENT` const
- use this getter to retrieve the call args
- verify call args content is equal to your inputs (`callArguments.put("index", 1);`)
- don't `return` and `Assert.fail();` you can simply use `assertEquals("message", expected, actual)`